### PR TITLE
[Merged by Bors] - chore: remove unnecessary `nolint simpNF`s

### PR DESCRIPTION
**Pull request by adomani**
This arose in the context of [this Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/dsimpNF/near/445455876).

See also
* #13889 for a previous hand-made version;
* #13970 for the automatic detection of unnecessary `nolint simpNF`s, (at least locally unnecessary).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
bors d+

**Comment by adomani** [comment]
CI, and in particular linting, was successful in #13984, so I think that it is fine to

bors r+

**Comment by adomani** [comment]
maintainer merge

maybe?  Bors does not seem to have picked up the delegation.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/Algebra/Free.lean [hunk 1]
```diff
@@ -169,7 +169,7 @@ protected def recOnPure {C : FreeMagma α → Sort l} (x) (ih1 : ∀ x, C (pure
 #align free_magma.rec_on_pure FreeMagma.recOnPure
 
 -- Porting note (#10675): dsimp can not prove this
-@[to_additive (attr := simp, nolint simpNF)]
+@[to_additive (attr := simp)]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Free.lean:172]: Can you follow up to remove all the comments that explain why we used `nolint simpNF`, like this one?

File: Mathlib/Algebra/MvPolynomial/Supported.lean [hunk 2]
```diff
@@ -55,15 +55,15 @@ noncomputable def supportedEquivMvPolynomial (s : Set σ) : supported R s ≃ₐ
     (AlgEquiv.ofInjective (rename ((↑) : s → σ)) (rename_injective _ Subtype.val_injective)).symm
 #align mv_polynomial.supported_equiv_mv_polynomial MvPolynomial.supportedEquivMvPolynomial
 
-@[simp, nolint simpNF]  -- Porting note: the `simpNF` linter complained about this lemma.
+@[simp]  -- Porting note: the `simpNF` linter complained about this lemma.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/MvPolynomial/Supported.lean:58]: Or this one

File: Mathlib/Algebra/Order/Ring/WithTop.lean [hunk 3]
```diff
@@ -58,7 +58,7 @@ lemma top_mul' : ∀ (b : WithTop α), ⊤ * b = if b = 0 then 0 else ⊤
 #align with_top.top_mul WithTop.top_mul
 
 -- eligible for dsimp
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Ring/WithTop.lean:60]: Or this one

**Pull request closed without merging**